src/test/java/bowling/FrameTest.java

```diff
@@ -0,0 +1,225 @@
+package bowling;
+
+import bowling.exception.BowlingException;
+import bowling.exception.FrameOverException;
+import bowling.exception.InvalidPinQuantityException;
+import org.junit.Test;
+
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class FrameTest {
+    @Test
+    public void normalFrameHasTwoRolls() throws BowlingException {
+        Frame frame = new Frame(false);
+
+        frame.roll(1);
+        assertThat(frame.isOver(), is(false));
+
+        frame.roll(1);
+        assertThat(frame.isOver(), is(true));
+    }
+
+    @Test
+    public void normalFrameScore() throws BowlingException {
```

Comment:
public void roll_WhenFrameIsRolledWith1_AndFrameIsRolledWith4_ThenScoreIs5

Reply:
Alternativelly it is possible to throw in comments in the test
```
public....
// given bla bla
x = ...
// when tested code
value = tested.func()
// then correct
assert(value)
```

But that can be quite terse. It is useful for integration and service tests. Necessary for e2es. Unit tests probably not necessary as unit tests are very dependant on implementation.